### DISCUSSION TITLE
best way to catch arguments

### DISCUSSION THREAD

**Discussion by vjonas** [google/zx#263] [Help needed] [answered]
Hey all

What would be the best way to catch variables passed in on the script.mjs file?
ex: 
zx script.mjs --environment dev
or
zx script.mjs dev

Cheers

**Comment by antonmedv** [comment]
It depends.

**Comment by vjonas** [comment]
Could you provide an example on how to get the parameter in the js code?

**Reply by antonmedv** [reply]
https://github.com/google/zx#minimist-package

**Reply by vjonas** [reply]
> https://github.com/google/zx#minimist-package

Is it as simple as:

await $`mkdir $0` ?

**Reply by antonmedv** [reply]
No.

**Reply by vjonas** [reply]
Any tips on how to manage that?

**Reply by antonmedv** [reply]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Comment by onedr0p** [comment, marked as answer]
You can do this something like:

```js
const action = argv["action"] || process.env.ACTION
const app = argv["app"] || process.env.APP
const namespace = argv["namespace"] || process.env.NAMESPACE

if (!app) { throw new Error("Argument --app or envirornment variable APP not set") }
if (!namespace) { throw new Error("Argument --namespace or envirornment variable NAMESPACE not set") }
```

`./thing.mjs --action blah ....`
